src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -915,13 +915,11 @@ private synchronized JsonElement readTemperatureProfileRca(String rca) {
       // made from a data node, it returns a 400 saying it can only be queried from the elected
       // master.
       if (rca.equals(ClusterTemperatureRca.TABLE_NAME)) {
-        Field<Integer> foreignKeyField = DSL.field(
-            SQLiteQueryUtils.getPrimaryKeyColumnName(ResourceFlowUnit.RCA_TABLE_NAME),
-            Integer.class);
-        SelectJoinStep<Record> query = SQLiteQueryUtils
-            .buildSummaryQuery(create, ClusterTemperatureSummary.TABLE_NAME,
-                mostRecentRecord.get(primaryKeyField),
-                foreignKeyField);
+        SelectJoinStep<Record> query = SQLiteQueryUtils.buildSummaryQuery(
+            create,
+            ClusterTemperatureSummary.TABLE_NAME,
+            mostRecentRecord.get(primaryKeyField),
+            primaryKeyField);
```

Comment:
Let's use the new Object persisting mechanism

Reply:
@yojs  : Can you add more details here? Let's create a separate Issue and work on it given this PR is old. @yujias0706

Reply:
Created issue: https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/559